invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/AccessField/components/MetadataAccess.js

```diff
@@ -12,15 +12,41 @@ import { i18next } from "@translations/invenio_rdm_records/i18next";
 import Overridable from "react-overridable";
 
 export const MetadataAccess = (props) => {
-  const { recordAccess, communityAccess } = props;
+  const {
+    recordAccess,
+    communityAccess,
+    record,
+    recordRestrictionGracePeriod,
+    allowRecordRestriction,
+  } = props;
   const publicMetadata = recordAccess === "public";
   const publicCommunity = communityAccess === "public";
 
+  const checkGracePeriod = () => {
+    const created = record?.created;
```

Comment:
if https://github.com/inveniosoftware/invenio-rdm-records/pull/1724/files#r1601326255 is applied, additional check if record exists is not needed